python-spec/src/somacore/data.py

```diff
@@ -158,9 +224,39 @@ def read(
         result_order: options.ResultOrderStr = _RO_AUTO,
         platform_config: Optional[options.PlatformConfig] = None,
     ) -> pa.Tensor:
-        """Reads the specified subarray from this NDArray as a Tensor.
-
-        TODO: Additional per-param documentation.
+        """Reads the specified subarray from this DenseNDArray as a Tensor.
+
+        Coordinates must specify a contiguous subarray, and the number of
+        coordinates must be less than or equal to the number of dimensions.
+        For example, if the array is 10×20, acceptable values of ``coords``
+        include ``()``, ``(3, 4)``, ``[slice(5, 10)]``, and
+        ``[slice(5, 10), slice(6, 12)]``.  Slice indices are doubly-inclusive.
+
+        :param coords: A per-dimension sequence of coordinates defining
+            the range to read.
+        :param batch_size: The size of batches that should be returned
+            from a read. See :class:`options.BatchSize` for details.
+            XXX How does this work if this returns a single Tensor?
```

Comment:
please spell `XXX` as `TODO` to make it a more consistently found `grep` target

Reply:
@thetorpedodog i saw it only b/c it was flagged in green by the GitHub PR tool

anyone coming along later won't have that benefit

my ask is that we not mix `XXX` and `TODO` when we could just spell `TODO`

this is an ask of just a few keystrokes